flink-runtime/src/main/java/org/apache/flink/runtime/operators/GroupReduceCombineDriver.java

```diff
@@ -170,7 +171,12 @@ public void run() throws Exception {
 
 			// write the value again
 			if (!this.sorter.write(value)) {
-				throw new IOException("Cannot write record to fresh sort buffer. Record too large.");
+				if (oversizedRecordCount == Long.MAX_VALUE) {
+					LOG.error("Number of oversized record has exceeded MAX Long");
+				} else {
+					++oversizedRecordCount;
+					LOG.warn("Cannot write record to fresh sort buffer. Record too large. Oversized record count: {}", oversizedRecordCount);
+				}
```

Comment:
I tried to call
```
this.output.collect(value)
```
I couldn't figure out how to convert this single value from type IN to type OUT. Could you guys please help me? @StephanEwen @rmetzger